lms/services/lms_api.py

```diff
@@ -0,0 +1,59 @@
+from typing import Literal, NotRequired, Protocol, TypedDict
+
+
+class APICallInfo(TypedDict):
+    path: str
+    authUrl: NotRequired[str]
+
+
+class LMSDocument(TypedDict):
+    """Represents a document or folder in an LMS's file storage."""
+
+    type: Literal["File", "Folder"]
+    mime_type: NotRequired[Literal["text/html", "application/pdf", "video"]]
+
+    id: str
+    """ID of the document in our sytem. Often in the form schema://DETAILS/"""
+
+    lms_id: str
+    """ID of the document in the LMS itself."""
```

Comment:
I noted above that the frontend code doesn't use this property. Is this data structure being used for other purposes in the backend?

Edit: Oh I see, it looks like this is also used to store a copy of file metadata in the LMS DB.